src/codegen.cpp

```diff
@@ -8731,6 +8731,7 @@ Error create_c_object_cache(CodeGen *g, CacheHash **out_cache_hash, bool verbose
     for (size_t arg_i = 0; arg_i < g->clang_argv_len; arg_i += 1) {
         cache_str(cache_hash, g->clang_argv[arg_i]);
     }
+    cache_bool(cache_hash, g->function_sections);
```

Comment:
This is good - it's for compiling C code. You should add the -ffunction-sections arg for c code here (look for -fPIC for example).

There's another place to add this bool to a cache hash for compiling zig code ( look for where those other bools from CodeGen struct are used)

Reply:
Ah yes, I see. Missed that other one. What about the g->function_sections being uninitialized, should I just set it to false in codegen_create?

Reply:
codegen_create zero intializes everything, so it's false by default